src/core/agent/records/connectionRecord.ts

```diff
@@ -9,14 +9,15 @@ interface ConnectionRecordStorageProps {
   oobi: string;
   groupId?: string;
   pending: boolean;
+  pendingDeletion?: boolean;
```

Comment:
We will never be adding a connection that's pending deletion. Remove this from the props and set `pendingDeletion = false;` as a default in the class